\documentclass[10pt]{beamer}
\usepackage{pgfpages}
\setbeameroption{show notes on second screen=right}
%\setbeameroption{show notes on second screen}
\usetheme{metropolis}
%\usepackage[sfdefault]{Fira Sans}
\usepackage{appendixnumberbeamer}
\setbeamercolor{background canvas}{bg=white}
\usepackage{booktabs}
\usepackage[scale=2]{ccicons}
\usepackage{textgreek}
\usepackage{ulem}
\usepackage{pgfplots}
\usepgfplotslibrary{dateplot}
\usepackage{multicol}
\usepackage{tikz}
\usepackage{listings,newtxtt}
\lstset{basicstyle=\ttfamily, keywordstyle=\bfseries}
\usepackage{xspace}
\newcommand{\themename}{\textbf{\textsc{metropolis}}\xspace}

%https://tex.stackexchange.com/questions/55589/text-justify-in-beamer
\usepackage{ragged2e}
\justifying
\apptocmd{\subtitle}{}{\justifying}{}
\usepackage[sort&compress]{natbib}
\usepackage{bibentry}
\usepackage{usebib}
\newbibfield{journal}
%\def\smi{library.bib}
\def\smi{out/ln/updir/mw-gcthesis-oral/library.bib}
\bibinput{library}
\nobibliography*

\usepackage{etoolbox}
\makeatletter
\apptocmd\beamer@framenotesbegin
{\centering}
{}{}
\makeatother



%\title{Thesis defence}
\title{Soutenance de thèse}
\subtitle{Décryptage de dynamiques épigénomiques au cours de la thymopoïèse et de la spermatogénèse en appliquant une méthodologie de recherche reproductible à des données de séquençage à haut débit}
%\subtitle{Methodological developments for reproducible research in bioinformatics and applications to the study of epigenetic regulation during thymopoieisis and spermatogenesis}

\date{\today}
\author{Guillaume Charbonnier}
\institute{TAGC}
% \titlegraphic{\hfill\includegraphics[height=1.5cm]{logo.pdf}}

\begin{document}

%\maketitle

\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/title.pdf}
  \includegraphics[width=1\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note{Bonjour à tous, 

  Merci ....

  Il y a un peu plus de 4 ans, j'ai commencé à travailler en tant que bioinformaticien au TAGC, et comme la plupart des gens de cette espèce, je m'intéresse à la meilleure façon de mener des analyses de données à partir d'outils en ligne de commande. 

  Sur le plan technique, l'objectif principal de ma thèse consistait à déterminer un ensemble d'outils et de méthodes permettant d'assurer la reproductibilité, l'automatisation, le partage, l'archivage, le développement simplifié, l'optimisation et la réutilisation d'analyses complexes de données volumineuses. Cet objectif générique a été couplé à des objectifs spécifiques de recherche fondamentale dans le domaine de l'épigénétique, et plus particulièrement dans l'étude de dynamiques épigénétiques de deux modèles biologiques : la thymopoïèse humaine et la spermatogénèse murine.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Ulirsch2019-hsc-lymphoid-myeloid-lineages.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{La majorité des cellules d'un organisme partagent le même génome. Pourtant différents types cellulaires expriment des programmes génétiques différents leur permettant d'assurer la spécificité de leur fonction, avec par exemple ici différents types cellulaires effecteurs du système immunitaire de mammifères.}
  % conservation des types cellulaires chez les mammifères par opposition aux autres animaux:
  % http://www.ipubli.inserm.fr/bitstream/handle/10608/4432/MS_1991_7_665.pdf
  % https://web.archive.org/web/20190916133519/http://www.ipubli.inserm.fr/bitstream/handle/10608/4432/MS_1991_7_665.pdf
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Dogan2018-fig1/chromosome-territories.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{En s'intéressant de plus près au noyau de ces cellules en interphase, on peut observer les différents chromosomes répartis en territoires distincts dans l'espace.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Dogan2018-fig1/ab-compartments.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{L'analyse de données d'interaction de type Hi-C permet de distinguer à l'intérieur de chaque chromosome des compartiments notés A et B qui sont respectivement actifs ou inactifs en fonction du type cellulaire. %associés à la chromatine ouverte et fermée respectivement.
  }
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Dogan2018-fig1/tads.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{A une échelle inférieure, on distingue les domaines topologiquement associés correspondants à des régions dont les différents sites sont plus fréquemment associés entre eux plutôt qu'avec ceux d'autres régions.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Dogan2018-fig1/chromatin-loops.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Ces fréquences d'interaction plus importantes pour certains sites sont la conséquence du repliement de l'ADN en boucles, repliement qui est contrôlé par la présence de complexes protéiques.}% comprenant généralement Polycomb et/ou CTCF.}
  %https://www.igh.cnrs.fr/fr/recherche/departements/dynamique-du-genome/21-chromatine-et-biologie-cellulaire}{IGH Giaocomo Cavalli
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Dogan2018-fig1/custom-layout.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{A une échelle encore inférieure, on distingue l'ADN nucléaire enroulé autour d'octamères d'histones pour former des nucléosomes. L'espacement entre nucléosomes est variable et peut être remodelé par des mécanismes cellulaires pour moduler le niveau de condensation et l'accessibilité de l'ADN.}
\end{frame}
%\begin{frame}[plain]
%  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Dulac2010-mechanisms-involved-in-chromatin-modifications.pdf}
%  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
%  \note{Sauter cette diapositive}
%\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig2/pluripotency-state.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{En regardant plus en détail un nucléosome, on peut distinguer les extrémités N-terminales des histones dépassant de la structure globulaire.%du nucléosome. 
  Les résidus composants ces extrémités sont accessibles à des enzymes de modifications qui ont, dans le cas présent, acétylés les lysines 9 et 27 de l'histone 3, ainsi que celles de l'histone 4, et méthylés la lysine 4 de l'histone 3.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig2/histone-deacetylation-and-demethylation.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Les processus inverses, de déacétylation et de déméthylation des lysines sont respectivement réalisés par des histones déacétylases, en abrégé HDAC, et par une déméthylase non représentée ici.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig2/local-heterochromatin.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{La méthyltransférase G9A va alors méthyler la lysine 9 de l'histone 3, ce qui va permettre la fixation du protéine à chromodomaine HP1. La fonction de ces protéines est d'agir comme répresseur des gènes dont ils sont la cible en entrainant le remodellage de la chromatine accessible en hétérochromatine inaccessible.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig2/turning-off-pluripotency-genes.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{L'ADN lui même peut aussi être sujet à des modifications enzymatiques. Plus particulièrement les cytosines suivies de guanine peuvent être méthylées par une enzyme de la famille des DNMT. La succession de ces différentes étapes de modifications de la chromatine correspond à la dynamique épigénétique observée au niveau des promoteurs de gènes de la pluripotence des cellules souches embryonnaires lorsque celles-ci s'engagent dans une voie de différentiation.}
  %Idée: aller jusqu'au 4 puis révéler que cette dynamique épigénétique correspond à l'inactivation de gènes pluripotents. 
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig4/1.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Si on s'intéresse maintenant au cas d'un autre promoteur, celui d'un gène A qui s'exprime spécifiquement dans les cellules de la lignée myéloïde. Ce gène est réprimé dans les cellules souches hématopoiétiques par le mécanisme de méthylation de l'ADN.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig4/2.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Son statut épigénétique est inchangé dans un progéniteur lymphoïde.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig4/3.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{En revanche le promoteur perd sa méthylation lorsque la cellule s'engage dans une voie de différentiation myéloïde, ce qui lève le verrou épigénétique empêchant l'expression du gène A.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig4/4.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{En regardant maintenant le cas du promoteur du gène B, spécifique de la lignée lymphoïde, celui-ci ne peut s'exprimer dans les cellules souches hématopoïétiques, non par le mécanisme de répression de méthylation de l'ADN, mais par celui de la tri-méthylation de la lysine 27 de l'histone 3.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig4/5.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Dans la lignée myéloïde, aucun changement.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig4/6.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Alors que dans le progéniteur lymphoïde, le verrou de la tri-méthylation est levé. Cet exemple très simplifié illustre pourquoi il est important de s'intéresser aux dynamiques épigénomiques dans différents modèles biologiques puisque des mécanismes différents sont à l'oeuvre dans la régulation des programmes d'expression.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig5-custom-layout/1.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Un dernier exemple classique: Les gènes bivalents. Ils possèdent à la fois la marque de répression H3K27me3 et celle d'activation H3K4me3. Afin d'affiner leur statut épigénétique, il est nécessaire d'avoir accès à d'autres modifications épigénétiques.}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig5-custom-layout/2.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{S'ils possèdent en plus des monométhylations sur les lysines 4, 9 et 20 de l'histone 3 ainsi que le variant d'histone H2AZ en remplacement de H2A, alors ce gène bivalent peut considéré comme activable. On peut noter au passage qu'il est tout a fait possible d'observer expérimentalement plusieurs modifications épigénétiques sur les mêmes résidus d'histones puisqu'il existe deux queues pour chaque type d'histone par nucléosome. D'autre part, les observations faites à partir de ChIP-Seq se font sur des pools de cellules qui peuvent avoir des états différents à un instant donné.}
  %Lire Tee2014 pour infos sur les symétries histoniques dans les nucléosomes.
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Cedar2011/fig5-custom-layout/3.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{Si on observe majoritairement la marque H3K9me3, le gène peut être considéré comme réprimé. Pour tous ces exemples d'introduction, la région des gènes considérés est leur promoteur...}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Ong2011.pdf}
  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{\smi}%
  \note{... Mais la connaissance de la présence de différentes modifications épigénétiques permet également de classifier l'ensemble du génome en éléments fonctionnels, tel que les enhancers et silenceurs. L'activation ou l'inactivation de ces éléments fonctionnels dans un contexte de différentiation cellulaire sain ou pathologique permet d'expliquer l'expression de programme génétique particuliers. L'étude des dynamiques épigénomiques qui gouvernent ou révèlent ces activations ou inactivations peut permettre de retracer l'enchainement des événements qui amènent une cellule à s'engager dans une voie de différentiation.}%
  %er quer l'idée que des mécanismes différents sont à l'oeuvre pour l'activation et l'inactivation de certains gènes. D'où l'intérêt d'étudier les mécanismes. Peut être retirer insulateur de la figure ?}
\end{frame}
%\begin{frame}[plain]
%  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/ElKennani2017-histone-variants-in-mouse.pdf}
%  \includegraphics[width=1\textwidth, height=.95\textheight, keepaspectratio]{\smi}
%\end{frame}
%\begin{frame}[plain]
%  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/ElKennani2017-histone-variants-in-mouse-h2al2-highlight.pdf}
%  \includegraphics[width=1\textwidth, height=.95\textheight, keepaspectratio]{\smi}
%\end{frame}
%\begin{frame}[plain]
%  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Huang2014-histones-ptms.pdf}
%  \includegraphics[width=\textwidth, height=\textheight, keepaspectratio]{\smi}
%\end{frame}
%\begin{frame}[plain]
%  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Huang2014-histones-ptms-reference-epigenome-highlight.pdf}
%  \includegraphics[width=\textwidth, height=\textheight, keepaspectratio]{\smi}%
%  \note{Tout comme dans l'exemple des gènes bivalents qui peuvent être classifiés plus précisément en gènes activables ou réprimés, on peut supposer que l'intégration de davantage de modifications épigénétiques permettent une définition plus précise de chaque région du génome.}%
%\end{frame}
%\begin{frame}[plain]
%  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Huang2014-histones-ptms-h4k5k8acbu-highlight.pdf}
%  \includegraphics[width=\textwidth, height=\textheight, keepaspectratio]{\smi}%
%  \note{Parler de H4K5ac/bu H4K8ac/bu ici et évoquer l'article. Conclure sur le travail titanesque de classifier fonctionnellement toutes les marques et variants: fonctions/interactions. Poser la problématique biologique de l'étude des dynamiques épigénomiques. Etude de ce grand schéma, code épigénétique}%
%\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/necker-collaboration.pdf}
  \includegraphics[width=\textwidth, height=.9\textheight, keepaspectratio]{\smi}%
  \note{Plus particulièrement, je vais vous présenter aujourd'hui une étude des dynamiques épigénomiques au cours de la thymopoïèse réalisée pendant ma thèse. Cette étude a pu être menée grâce aux données épigénétiques produites dans le cadre du projet Européen Blueprint, avec un soupçon d'Encode, un zest de Roadmap, et complémentées par des données générées hors consortiums issus d'échantillons thymiques de donneurs sains (et de patients?) de l'institut Necker.}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Ulirsch2019-fig1-recolored.pdf}
  \includegraphics[width=\textwidth, height=\textheight, keepaspectratio]{\smi}%
  \note{Sur cette vue d'ensemble de l'hématopoïèse, la thymopoïèse est délimitée par la surface en bleu et correspond à la différentiation de progéniteurs lymphoïdes communs, noté ici CLP, en lymphocyte T CD4+ ou CD8+.}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/thymus/1.pdf}
  \includegraphics[width=\textwidth, height=\textheight, keepaspectratio]{\smi}%
  \note{}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/thymus/2.pdf}
  \includegraphics[width=\textwidth, height=\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/thymus/3.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/thymus/4.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/thymus/5.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/thymus/6.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]{Objectives}
  \begin{itemize}
    \item Replace thymic subpopulations in the epigenomic landscape of hematopoiesis
    \item Shed light on epigenomic dynamics during thymopoiesis with a focus on distal regulatory regions and enhancers
  \end{itemize}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/wgbs/matrix.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note{Hello}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/wgbs/matrix-tsne.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note[item]{Retrieve CpG methylation calls for all 135 Blueprint hematopoietic samples as bigwig files}%
  \note[item]{Convert calls to matrix}%
  \note[item]{Remove CpG overlapping repeats from repeatMasker, or MXY chromosomes}%
  \note[item]{Perform t-SNE on matrix}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/matrix-model.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note{Définir region = dynamic region. A réfléchir}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/matrix-mca.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/msigdb-model.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/hematopoiesis/hsc-marker.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/hematopoiesis/hsc-t-markers.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/hematopoiesis/b-marker.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/hematopoiesis/b-myeloid-markers.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/both-mca.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/riverplot-model.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/riverplot-heatmap.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/t-lineage/1.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/t-lineage/2.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/t-lineage/3.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/t-lineage/4.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/t-lineage/5.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/t-lineage/6.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/hypometh-clusters/1.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/hypometh-clusters/2.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note{Chaine de production des données semble cohérente avec ce que l'on attend de la variance entre les échantillons. On peut donc passer à une analyse exploratoire des jeux de données.}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/hypometh-clusters/3.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/hypometh-clusters/4.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/hypometh-clusters/5.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/hypometh-clusters/6.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/h3k27ac-clusters/1.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note{Discussion sur la notion de pic, artefacts visibles dans la dynamique}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/h3k27ac-clusters/2.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/h3k27ac-clusters/3.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/h3k27ac-clusters/4.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/h3k27ac-clusters/5.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/enhancers/1.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/enhancers/2.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/enhancers/3.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/atac.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/atac-gobp.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/atac-motifs.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/atac-wgbs.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/atac-wgbs-h3k27ac.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/h3k27ac.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/rnaseq-h3k27ac-no-casero.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/rnaseq-h3k27ac-with-casero.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/casero-violin-barplot.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/atac-clusters/casero-violin-barplot-top-fc.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/casero-dear.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/chromatin-states/genome-view/tcra.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/ea-tf-mouse.pdf}
  \includegraphics[width=\textwidth, height=.85\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/rna-clusters/all.pdf}
  \includegraphics[width=\textwidth, height=.9\textheight, keepaspectratio]{\smi}
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/rna-clusters/c13-tf.pdf}
  \includegraphics[width=\textwidth, height=.65\textheight, keepaspectratio]{\smi}
\end{frame}

\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/tall/H3K27ac_HOXA5-9_TLX1_TLX3_ealpha_barplot.png}
  %\def\smi{out/ln/updir/mw-gcthesis-oral/ink/tall/H3K27ac_HOXA9_TLX1_TLX3_ealpha_barplot_recolored_trimmed.png}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note{Classes}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/tall/H3K27ac_HOXA5-9_TLX1_TLX3_ealpha_barplot.png}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \begin{tikzpicture}[overlay, remember picture]
    \node at (current page) 
    {
      \def\smi{out/ln/updir/mw-gcthesis-oral/ink/tall/ealpha_h3k27ac_groups.pdf}
      %\def\smi{out/ln/updir/mw-gcthesis-oral/ink/tall/dot-box-violin-plot-H3K27ac-ealpha.pdf}
      \includegraphics[width=\textwidth, height=.65\textheight, keepaspectratio]{\smi}
    };
  \end{tikzpicture}
  \note{Boxplot}%
\end{frame}
\begin{frame}[plain]
  \def\smi{out/ln/updir/mw-gcthesis-oral/ink/Highlight_HOXA_Blueprint_paper.pdf}
  \includegraphics[width=\textwidth, height=.95\textheight, keepaspectratio]{\smi}%
  \note{Classes}%
\end{frame}

%\begin{frame}{Key results}
%  \begin{itemize}
%    \item A reference epigenome of human early T cell differentiation has been generated
%      \begin{itemize}
%        \item Thymic subpopulations exhibit consistent epigenomic landscape when comparing between them, and between all HSC lineages
%        \item Differentiation from HSC to mature T-cell lineage is associated with a shift from enhancers and transcribed regions to heterochromatin
%          %  \end{itemize}
%        \item DNA demethylation is a hallmark of distal regulatory elements irrespective of their activation status in T cells
%          \begin{itemize}
%            \item Distal hypomethylated regions are predominantly (77\%) constitutively demethylated accross thymic subpopulations
%            \item Dynamic distal hypomethylated regions are associated with higher level of H3K27ac and H3K4me1, and higher expression level of nearest genes
%            \item Distal regions losing methylation during T-cell differentiation are associated with T-cell functions whereas constitutively hypomethylated regions are mainly associated with other hematopoietic functions
%          \end{itemize}
%        \item Chromatin opening precedes enhancer activation
%        \item The TCRA enhancer (Eα) is in an open but epigenetically silent configuration in immature thymocytes
%        \item The HOXA locus is progressively repressed during early T cell differentiation
%        \item HOXA5-9 proteins repress Eα activity via their homeodomain
%        \item HOXA9 overexpression imposes developmental bias towards γδ T cell lineage
%        \item Homeodomain protein deregulation leads to TCRγδ bias in human T-ALL
%  \end{itemize}
%\end{frame}








%\begin{frame}{Table of contents}
%  \setbeamertemplate{section in toc}[sections numbered]
%  \small
%  \tableofcontents[hideallsubsections]
%\end{frame}
%\section{Context of my thesis}
%\begin{frame}{Summary of contributions}
%  \begin{itemize}
%      %\item \bibentry{Goudarzi2016}
%    \item \def\key{Goudarzi2016}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item \def\key{Dao2017}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item \def\key{Barral2017}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item \def\key{Shiota2018}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item \def\key{Lopez2019}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item Paper in preparation with Necker
%    \item \textit{greatr: Better multi-sample visualization for predictions of cis-regulatory regions.} \textbf{\href{https://github.com/guillaumecharbonnier/greatr}{github}}
%    \item \textit{MetaWorkflow: A general purpose algorithmical methodology for workflow development.} \textbf{\href{https://github.com/guillaumecharbonnier/mw}{github}}
%  \end{itemize}
%\end{frame}
%\begin{frame}{Summary of activities}
%  \begin{itemize}
%    \item \citet{Goudarzi2016} \textbf{ChIP-Seq}
%    \item \citet{Dao2017} \textbf{CapStarr-Seq, ChIP-Seq}
%    \item \citet{Barral2017} \textbf{MNase-Seq}
%    \item \citet{Shiota2018} \textbf{ChIP-Seq, RNA-Seq, Microarrays}
%    \item \citet{Lopez2019} \textbf{Development, Python}
%    \item Paper in preparation with Necker \textbf{ChIP-Seq, ATAC-Seq, WGBS, RNA-Seq}
%    \item greatr \textbf{Development, R}
%    \item mw \textbf{Development, misc languages}
%  \end{itemize}
%\end{frame}
%\begin{frame}{Common denominator}
%  \centering
%  Bioinformatics + Data analysis + Command line + Reproducibility % Reproducible research
%  % Reproducible data analysis
%\end{frame}
%\begin{frame}{Adjusted definition for the scope of this presentation}
%  \centering
%  Reproducibility is the ability to recompute data analytic results given an observed dataset and knowledge of the data analysis pipeline.
%  %\begin{itemize}
%  %  \item Reproducibility is the ability to recompute data analytic results given an observed dataset and knowledge of the data analysis pipeline.
%  %    %\item Flexibility is the ability to change or adjust to cope with variable circumstances.
%  %    % Scalability + Easy chaining and bypassing rules.
%  %  %\item Bioinformatics is the science of producing biological interpretation and
%  %    %publication-ready
%  %   % plots from data \textit{using a chaining of command-line tools and scripts}.
%  %\end{itemize}
%\end{frame}
%\begin{frame}{Reproducibility in theory}
%  \centering
%  YES or NO
%\end{frame}
%%http://ropensci.github.io/reproducibility-guide/sections/introduction/
%\begin{frame}{Reproducibility in practice}
%  A wide spectrum of practices to implement:
%  \begin{itemize}
%    \item Literate computing%, authoring, and publishing % Jupyter, knitr, 
%    \item Data tracking
%    \item Automation
%    \item Portability %Software deployment
%    \item Version control %Collaboration
%    \item Continuous integration
%    \item Immutability
%    \item Scalability
%    \item Reentrability
%    \item Licensing
%    \item Sharing % Distribution % Documentation
%    \item Reusability
%    \item Simplicity
%    \item Modularity
%  \end{itemize}
%\end{frame}
%\begin{frame}{Key points in bioinformatics}
%  \begin{itemize}
%    \item Big raw files and huge stack of processed files.
%    \item Computation-heavy needs but highly scalable and parallelizable tasks.
%    \item Complex analyses with high degree of freedom in choices for tools and parameters.
%    \item Florishing area of research with fast obsolescence in technologies repercuting on software and analysis approaches.
%    \item More and more public available data which could be used for data integration of various approaches.
%    \item Widespread open-source communities.
%  \end{itemize}
%\end{frame}
%\begin{frame}{Requirements for reproducible research methodology}
%  It should provide a quick and easy way:
%  \begin{itemize}
%    \item to reproduce a complete published analysis in one command on any POSIX system without admin permissions.
%    \item to identify which file is mandatory for archiving.
%    \item to select and remove all files that are not mandatory and could be removed to free memory space.
%    \item to code for scalable and parallelized analyses.
%    \item to benchmark various tools and parameters and store their outputs at the same time.
%    \item to handle various types of analyses in a single organized directory.
%    \item to develop indifferently draft, elegant code or dirty hack without fearing for conflicts.
%  \end{itemize}
%\end{frame}
%\begin{frame}{A popular combination of tools to implement a solution}
%  \begin{itemize}
%      %\item \bibentry{Goudarzi2016}
%    \item \def\key{Koster2012}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item \def\key{Dale2018}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item \def\key{Kurtzer2017}\citeauthor{\key} \textit{\usebibentry{\key}{title}}. \textbf{\usebibentry{\key}{journal}}. \citeyear{\key}.
%    \item git
%    \item travis/circleci
%    \item LaTeX/R-Markdown/reStructuredText
%  \end{itemize}
%\end{frame}
%%La reproductibilité en pratique : méthodes et outils
%%¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
%%
%%La reproductibilité de la recherche est un sujet qui prend une importance croissante dans notre environnement. Elle est un enjeu majeur pour l'intégrité scientifique. Dans un monde toujours plus numérique, chercheurs et informaticiens doivent collaborer à la mise en œuvre de méthodes et outils qui garantiront ce principe.
%%
%%Les données tout d'abord sont au coeur de la reproductibilité. Leur collecte, leur stockage, leur transformation, le contrôle de leur qualité, leur partage, leur identification sont sont
%%des préoccupations majeures.
%%
%%Les logiciels développés ainsi que les logiciels utilisés doivent également être gérés avec soins et recensés avec minutie. Quels outils et quels processus faut-il mettre en oeuvre pour garantir la précision et la reproductibilité de leur exécution ?
%%Comment automatiser l'interaction entre les logiciels et les données, selon quels workflows, avec quels outils et sur quelles infrastructures ?
%%
%%Enfin se posent également les question de l'organisation du travail au quotidien pour historiser toutes les actions de recherche : développement, exploration, etc. Avez-vous mis en place des outils pour faciliter cette traçabilité ?
%%
%%En tant qu'informaticiens, venez nous présenter les infrastructures, les outils et les processus que vous avez mis en oeuvre pour permettre la reproductibilité des résultats.
%%
%%Venez partager vos connaissances, et vos expériences en nous proposant des exposés de 20 à 50 min avant le 26/02/2018 à l'adresse suivante : <comite.aramis@listes.resinfo.org>
%%
%%Nous réserverons également un créneau de présentation pour quelques lightning talks (courte présentation de 5 min, 1-2 transparents, inscription dès à présent).
%%
%%Quelques suggestions de thèmes et mots-clefs :
%%
%%- Données : protection, qualité, traçabilité
%%- Stockage : sécurité, sauvegarde, archivage
%%- Code : Langage, versioning, portage
%%- Immutabilité : Conteneurisation (Docker, Singularity)
%%- Workflow : Intégration continue, livraison continue
%%- Sécurité : protection du potentiel scientifique
%%- Notebook : Jupyter, Rstudio
%%- Initiatives : Software Heritage
%%\begin{frame}{Solution proposal key points.}
%%  \begin{itemize}
%%    \item Snakemake: A make-like workflow manager which handle both local and cluster execution in a scalable and readable manner.
%%    \item Conda: A system-independent package manager.
%%    \item A new set of (trivial?) coding conventions to meet requirements.
%%  \end{itemize}
%%\end{frame}
%%\begin{frame}
%%  \frametitle{Straighforward workflow writing with scalability thanks to Snakemake.}
%%  \begin{columns}[c] % The "c" option specifies cente
%%    \column{.7\textwidth} % Left column and width
%%    \begin{itemize}
%%      \item A workflow consists of a set of rules which take input files and produce output files.
%%      \item An implicit naming convention allows Snakemake to chain rules together when asked for the wanted output file.
%%    \end{itemize}
%%    \centering
%%    The `latex' rule\\
%%    \includegraphics[width=1\textwidth, height=0.45\textheight, keepaspectratio]{doc/2017_04_28_gc_screenshots_for_guidelines/rule_latex.png}
%%    \column{.3\textwidth} % Right column and width
%%    \centering
%%    %{out/snakemake/dag/guidelines_conda_snakemake.pdf} \\
%%    Rulegraph for this presentation
%%    \includegraphics[width=1\textwidth, height=0.65\textheight, keepaspectratio]{out/graphviz/dot/snakemake/stdout_--rulegraph_--forceall_broad-analysis.pdf}
%%    %out/snakemake/rulegraph/guidelines_conda_snakemake.pdf}
%%  \end{columns}
%%\end{frame}
%\begin{frame}
%  \frametitle{Ensuring environment and tool support on any computer thanks to Conda}
%  \begin{itemize}
%    \item \href{https://conda.io/docs/}{Conda}: a package, dependency and environment management for any language and system.
%    \item Started as a \href{https://pypi.python.org/pypi/pip}{pip}+\href{https://virtualenv.pypa.io/en/stable/}{virtualenv equivalent}.
%    \item Grown up with alternatives to \href{https://www.cpan.org}{CPAN}, \href{https://cran.r-project.org}{CRAN}, \href{https://www.ctan.org}{CTAN} and \href{https://www.bioconductor.org}{Bioconductor}.
%    \item Developed into a system-independent \href{https://doc.ubuntu-fr.org/apt}{apt}/\href{https://doc.fedora-fr.org/wiki/YUM_:_Configuration_du_gestionnaire_de_paquets}{yum}/\href{https://doc.fedora-fr.org/wiki/DNF,_le_gestionnaire_de_paquets_de_Fedora}{dnf} equivalent.
%    \item Comparison versus \href{https://www.docker.com/}{Docker}/\href{http://singularity.lbl.gov/}{Singularity}:
%      \begin{itemize}
%        \item Killer feature: No root access required for installation.
%        \item Weakness: Uncorrectly builded packages may not work everywhere.
%          %\item Solution: You can propose a tested container where everything is working.
%      \end{itemize}
%  \end{itemize}
%\end{frame}
%\begin{frame}{Snakemake ``pull'' principle}
%  \includegraphics[width=1\textwidth, height=0.85\textheight, keepaspectratio]{out/wget/https/screenshotscdn.firefoxusercontent.com/images/20437834-bea9-42ce-87d8-d89d9846d67e.png}
%\end{frame}
%\begin{frame}{Best-practice workflow development paradigm: One task/analysis per workflow}
%  \begin{columns}[c] % The "c" option specifies cente
%    \column{.7\textwidth} % Left column and width
%    \begin{itemize}
%      \item \href{https://github.com/snakemake-workflows/rna-seq-star-deseq2}{rna-seq-star-deseq2}
%      \item \href{https://github.com/snakemake-workflows/dna-seq-gatk-variant-calling}{dna-seq-gatk-variant-calling}
%      \item \href{https://github.com/snakemake-workflows/accel-amplicon-trimming}{accel-amplicon-trimming}
%      \item \href{https://github.com/snakemake-workflows/single-cell-rna-seq}{single-cell-rna-seq}
%      \item \href{https://github.com/snakemake-workflows/single-cell-drop-seq}{single-cell-drop-seq}
%      \item \href{https://github.com/SnakeChunks/SnakeChunks}{SnakeChunks}
%      \item \href{https://github.com/sequana/sequana}{Sequana}
%    \end{itemize}
%    \column{.3\textwidth} % Left column and width
%    \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{out/graphviz/dot/ln/updir/mw/src/dot/rna-seq-star-deseq2_rulegraph.png}
%  \end{columns}
%\end{frame}
%\begin{frame}{Main difficulty for using Snakemake and other ``pull'' workflow managers}
%  \begin{itemize}
%    \item As a developer, you have the responsibility to find naming patterns for all the rules in your workflow. If your workflow grows in size, you may face ``ambiguous rules'' issues.
%    \item Discrepancies in naming patterns from different workflows are the main cause of hassle when trying to merge features from both.
%  \end{itemize}
%\end{frame}
%\begin{frame}[fragile]{Example of ``ambiguous rules''}
%  \begin{lstlisting}[language=Python]
%  rule gzip:
%    input: 
%      "{a_file_you_want_to_compress}"
%    output:
%      "{a_file_you_want_to_compress}.gz"
%    shell:
%      "gzip -f {input}"
%
%  rule wget_gz_from_somewhere_else:
%    output:
%      "wget/{a_url_leading_to_gz}"
%
%  rule deepTools_computeMatrix:
%    output:
%      "deepTools/{deepTools_matrix_are_gz}.txt.gz"
%  \end{lstlisting}
%\end{frame}
%\begin{frame}[fragile]{Snakemake built-in keyword solution limits modularity}
%  \begin{lstlisting}[language=Python]
%  ruleorder: deepTools_computeMatrix > wget > gzip
%  
%  rule gzip:
%    input: 
%      "{a_file_you_want_to_compress}"
%    output:
%      "{a_file_you_want_to_compress}.gz"
%
%  rule wget:
%    output:
%      "wget/{a_url_leading_to_gz}"
%
%  rule deepTools_computeMatrix:
%    output:
%      "deepTools/{deepTools_matrix_are_gz}.txt.gz"
%  \end{lstlisting}
%\end{frame}
%\begin{frame}[fragile]{Snakemake built-in keyword solution can't solve circular issues}
%  \begin{lstlisting}[language=Python]
%  rule gunzip:
%    input:
%      "{a_file_you_want_to_uncompress}.gz"
%    output:
%      "{a_file_you_want_to_uncompress}"
%
%  rule gzip:
%    input: 
%      "{an_unrelated_file_you_want_to_compress}"
%    output:
%      "{an_unrelated_file_you_want_to_compress}.gz"
%  \end{lstlisting}
%\end{frame}
%\begin{frame}[fragile]{Limiting scope of rules generate a lot of redundancy in code}
%  \begin{lstlisting}[language=Python]
%  rule gunzip_scope1:
%    input:
%      "a_file_you_want_to_uncompress.gz"
%    output:
%      "a_file_you_want_to_uncompress"
%  
%  rule gzip_scope1:
%    input:
%      "an_unrelated_file_you_want_to_compress"
%    output:
%      "an_unrelated_file_you_want_to_compress.gz"
%
%  rule gzip_scope2:
%    input:
%      "another_file_you_want_to_compress"
%    output:
%      "another_file_you_want_to_compress.gz"
%  \end{lstlisting}
%\end{frame}
%\begin{frame}{An other solution is possible...}
%\end{frame}
%\section{MetaWorkflow: a methodological experiment to find conventional algorithmical way for naming patterns}
%\begin{frame}{MetaWorkflow: a methodological experiment to find conventional algorithmical way for naming patterns}
%  \begin{itemize}
%    \item How to ensure no ambiguity can arise between rules while maintaining large scope for rules?
%    \item Is it possible to use Snakemake with an alternative paradigm ``One workflow for all analyses''?
%    %\item Transforming weakness into strength.
%    \item Is it possible to store a bioinformatician thesis project, a whole research career, or a project of any size with multiple collaborators inside a reproducible, fast and lightweight workflow?
%      %\item Simple solution: 
%  \end{itemize}
%\end{frame}
%%\begin{frame}{All rules and tools can be coerced to one of these rule types}
%%  \includegraphics[width=1\textwidth, height=0.75\textheight, keepaspectratio]{doc/2017_05_03_gc_rule_types/drawing.pdf}
%%\end{frame}
%\begin{frame}
%  \frametitle{All rules can and should be coerced to write their outputs in their own unique directory}
%  \begin{itemize}
%    \item out/\{tool-name\}/\{optional-subfunction\}\_\{arg1name\}-\{arg1value\}\_\{arg2name\}-\{arg2value\}\_\{\ldots\}/\{filler\}.\{ext\}
%    \item where \{filler\} is the main input file path from out level without its extension.
%    \item Example:
%      \begin{itemize}
%        \item Raw sequences:\\
%          \textcolor{red}{inp/fastq/sample}.fastq
%        \item Trimmed sequences:\\
%          out/\textcolor{blue}{sickle/se\_q-30\_t-sanger}/\textcolor{red}{inp/fastq/sample}.fastq
%        \item Aligned sequences:\\
%          out/\textcolor{green}{bowtie2/se\_x-GRCh38}/\textcolor{blue}{sickle/se\_q-30\_t-sanger}/\textcolor{red}{inp/fastq/sample}.sam
%      \end{itemize}
%  \end{itemize}
%\end{frame}
%\begin{frame}[fragile]{An example of rule for tools that have to be coerced to write to their own unique directory:\\Tools writing output file where input file is located}
%  \begin{lstlisting}[language=Python]
%  rule samtools_index:
%    input:
%      bam="out/{filler}.bam",
%    output:
%      bam="out/samtools/index/{filler}.bam",
%      bai="out/samtools/index/{filler}.bam.bai"
%    shell:
%      "ln -srf {input.bam} {output.bam}; "
%      "samtools index {output.bam}"
%  \end{lstlisting}
%\end{frame}
%\begin{frame}[fragile]{Another example of rule for tools that have to be coerced to write to their own unique directory:\\Tools writing output file where they are executed}
%  \begin{lstlisting}[language=Python]
%  rule thisTool
%    input:
%      "{filler}"
%    output:
%      "out/thisTool/{filler}"
%    shell:
%      """
%      WDIR=`pwd`
%      OUTDIR=`dirname {output}`
%      cd $OUTDIR
%      thisTool $WDIR/{input}
%      """
%  \end{lstlisting}
%\end{frame}
%\begin{frame}{Benefits of the ``One workflow for all analyses'' paradigm}
%  %  Motivations:
%  %  \begin{itemize}
%  %    \item How to write a Snakemake workflow that is able to hold
%  %    \item Can a bioinformatician thesis project, and by extent a whole research careeer, be stored as a reproducible, fast and lightweight workflow?
%  %    \item How a bioinformatician thesis project, and by extent a whole research careeer, can be stored as a reproducible, fast and lightweight workflow?
%  %  \end{itemize}
%  \begin{itemize}
%    \item Straightforward and computationally faster for:
%      \begin{itemize}
%        \item Data integration from various sources and types 
%        \item Benchmarking of tools and parameters
%      \end{itemize}
%    \item Easier analysis development and maintenance:
%      \begin{itemize}
%        \item Grant your analysis the ability to self-deploy using only your Snakefile
%        \item Limit redundancy of code for tools shared by multiple analyses
%        \item Prevent ``ambiguous rules''
%        \item No thinking about naming files and rules
%      \end{itemize}
%    \item Higher traceability of all files:
%      \begin{itemize}
%        \item Filepaths describe contents without ambiguity
%        \item Workflow reproducibly tracks ``formatting'' operations pre and post analysis.
%      \end{itemize}
%  \end{itemize}
%\end{frame}
%\begin{frame}{Drawbacks of the ``One workflow for all analyses'' paradigm}
%  \begin{itemize}
%    \item Browsing out directory require bioinformatics knowledge about tools or known paths.
%    \item Long file names make results hard to browse without terminal, and impossible in MS Windows before Windows 10 Anniversary Update when longer than 260 characters.
%    \item The workflow is dumb and makes no choice by itself. User has responsibility for all choices when asking for an output filepath.
%  \end{itemize}
%  $\rightarrow$ Efficient post-process solutions can be found with biologists (mw-sst project).
%\end{frame}
%\begin{frame}{Drawbacks of the ``One workflow for all analyses'' paradigm}
%  \begin{itemize}
%    \item R devices may trim long relative file names without warning: R ends without error but Snakemake can't find output files were they were specified.
%  \end{itemize}
%  $\rightarrow$ Moving to output directory and providing basename to R devices solves the issue.
%  \begin{itemize}
%    \item Privacy/Sharing issues when rules and files from different projects lie in the same repository
%  \end{itemize}
%  $\rightarrow$ Public sharing of core and common functions in ``mw'' and keeping project-specific parts in private repositories until publication.
%\end{frame}
%%\begin{frame}{Alternative paradigm: One workflow for all analyses}
%%  How?%
%%  \begin{itemize}
%%    \item Straightforward and computationally faster for:
%%      \begin{itemize}
%%        \item Data integration from various sources and types 
%%        \item Benchmarking of tools and parameters
%%      \end{itemize}
%%    \item Higher traceability of input files 
%%    \item Easier analysis developpment and maintenance:
%%      \begin{itemize}
%%        \item Limit redundancy of code for tools shared by multiple analyses
%%        \item Prevent ``ambiguous rules''
%%        \item No thinking about naming files and rules
%%      \end{itemize}
%%  \end{itemize}
%%\end{frame}
%\begin{frame}[fragile]{MetaWorkflow current state}
%  \begin{itemize}
%    \item More than 150 tool/subfonction pairs implemented as large scope rules in core repository \url{https://github.com/guillaumecharbonnier/mw}
%    \item 7 plugins for project specific components, publically available for published ones, \textit{e.g.} \url{https://github.com/guillaumecharbonnier/mw-broad}
%    \item 2 happy active users
%    \item Continuous integration deployment in progress
%    \item Open to contribution, issues and suggestions
%    \item To reproduce this presentation, install conda, git and snakemake then:
%  \end{itemize}
%    %basicstyle=\footnotesize, 
%  \begin{lstlisting}[language=Bash, basicstyle=\small, linewidth=1.1\textwidth, columns=fullflexible, breaklines=true, postbreak=\mbox{\textcolor{red}{$\hookrightarrow$}\space}]
%    git clone https://github.com/guillaumecharbonnier/mw
%    cd mw
%    snakemake out/tectonic/ln/updir/mw/src/tex/beamer/mw.pdf --use-conda
%  \end{lstlisting}
%\end{frame}
%\section{Epigenomic dynamics of early T cell progenitors reveals $TCR\alpha$ enhancer silencing by HOXA}
%\subsection{A reference epigenome of human early T cell differentiation}
%\begin{frame}{A reference epigenome of human early T cell differentiation}
%\end{frame}
%\begin{frame}{New high-throughput sequencing data generated for the Blueprint project}
%  \begin{itemize}
%    \item New samples: Figure S1.
%    \item Experiments: 6 main histone marks, WGBS, RNA-Seq, ATAC.
%  \end{itemize}
%\end{frame}
%\begin{frame}{Thymic subpopulations integrate consistently the epigenomic landscape of human hematopoiesis}
%  \begin{itemize}
%    \item S1B + 1A
%    \item \url{file:///home/gcharbonnier/mw/out/graphviz/dot/snakemake/stdout_--rulegraph_thymus-mca.pdf}
%    \item \url{file:///home/gcharbonnier/mw/out/graphviz/dot/snakemake/stdout_--dag_thymus-mca.pdf}
%  \end{itemize}
%\end{frame}
%\begin{frame}{Functional enrichment analysis of active-enhancer state consistently associate each population to their functions.}
%  \begin{itemize}
%    \item 1B
%  \end{itemize}
%\end{frame}
%\begin{frame}{Lineage specific genes displayed the expected chromatin dynamics across hematopoietic populations}
%  \begin{itemize}
%    \item 1C
%  \end{itemize}
%\end{frame}
%\begin{frame}{Chromatin states separate the main stages of thymic T cell differentiation from HSC and peripheral T cells}
%  \begin{itemize}
%    \item 1D
%    \item S1C:  V(D)J recombination VS cell homeostasis and activation functions?
%  \end{itemize}
%\end{frame}
%\begin{frame}{Chromatin state transitions during T-cell differentiation suggests a progressive loss of plasticity}
%  \begin{itemize}
%    \item 1E
%  \end{itemize}
%\end{frame}
%\begin{frame}{Known master transcription factors are consistently activated or repressed during T Cell commitment and differentiation}
%  \begin{itemize}
%    \item 1F
%  \end{itemize}
%\end{frame}
%\subsection{DNA demethylation marks distal regulatory elements irrespective of their activation status in T cells}
%\begin{frame}{DNA demethylation marks distal regulatory elements irrespective of their activation status in T cells}
%\end{frame}
%\begin{frame}{DNA methylation allows to retrace the lineages of the different hematopoietic samples}
%  \begin{itemize}
%    \item S3
%  \end{itemize}
%\end{frame}
%\begin{frame}{Constitutively hypomethylated distal regions represent a majority (77\%)
%  of all distal hypomethylated regions}
%    \begin{itemize}
%      \item 2A
%    \end{itemize}
%  \end{frame}
%  \begin{frame}{Constitutively hypomethylated distal regions are poorly enriched in H3K27ac, although level of hypomethylation is correlated with level of H3K27acetylation}
%    \begin{itemize}
%    \item 2B
%  \end{itemize}
%\end{frame}
%\begin{frame}{Constitutively hypomethylated distal regions are close to genes expressed at low level}
%  \begin{itemize}
%    \item 2C
%  \end{itemize}
%\end{frame}
%\begin{frame}{Constitutively hypomethylated distal regions are close to genes associated with genes related to hematopoietic lineages other than T Cells or not specifically enriched in relevant biological processes, although cluster 5 is significant for both T Cell related terms}
%  \begin{itemize}
%    \item 2D
%  \end{itemize}
%\end{frame}
%\begin{frame}{Dynamically hypomethylated distal regions are highly enriched in H3K27ac and close to genes highly expressed and related to T Cells}
%  \begin{itemize}
%    \item 2A + 2B + 2C + 2D
%  \end{itemize}
%\end{frame}
%\begin{frame}{Dynamically H3K27-acetylated and constitutively hypomethylated distal regions represents 17\%
%  of all distal hypomethylated regions}
%  \begin{itemize}
%    \item S3 + 2E + 2F
%  \end{itemize}
%\end{frame}
%\begin{frame}{Changes of H3K27ac levels roughly follow the expression dynamics of neighbor genes}
%  \begin{itemize}
%    \item 2E + 2G
%  \end{itemize}
%\end{frame}
%\begin{frame}{Three loci suggesting hypomethylation marks distal regulatory elements independently of their activation status}
%  \begin{itemize}
%    \item 2H
%  \end{itemize}
%\end{frame}
%\subsection{Chromatin opening precedes enhancer activation}
%\begin{frame}{Chromatin opening precedes enhancer activation}
%\end{frame}
%\begin{frame}{ATAC identifies 2084 and 2905 distal open chromatin regions in tCD34 and EC, respectively, with 1108 shared between thymic populations}
%  \begin{itemize}
%    \item 3A
%  \end{itemize}
%\end{frame}
%\begin{frame}{ATAC identifies 2084 and 2905 distal open chromatin regions in tCD34 and EC, respectively, with 1108 shared between thymic populations}
%  \begin{itemize}
%    \item 3A
%  \end{itemize}
%\end{frame}
%\begin{frame}{Who needs ggplot2 when we have Excel?}
%  \begin{itemize}
%    \item Functional enrichment analysis:3B
%    \item Motif enrichment analysis: 3B' (Credit to Quentin Ferré for gtftk peak\_anno)
%  \end{itemize}
%\end{frame}
%\begin{frame}{Open chromatin distal regions are constitutively demethylated independently of their opening timing}
%  \begin{itemize}
%    \item 3C
%    \item WGBS profiles may hide a small subset of of open chromatin regions that are not demethylated. Hence I propose as an additional argument to cluster this heatmap:\\
%      \url{out/deepTools/plotHeatmap_sortRegions-descend_sortUsing-region_length_averageTypeSummaryPlot-mean_missingDataColor-1_colorList-blueCyanYellowOrangeRed_heatmapHeight-28_heatmapWidth-3_whatToShow-hc_xAxisLabel-peak-center_refPointLabel-0_boxAroundHeatmaps-no/deepTools/computeMatrix_reference-point_referencePoint-center_beforeRegionStartLength-2000_afterRegionStartLength-2000_binSize-200_bed-hg38-atac-thymus_bw-hg38-BS-thymus-110-91.pdf}\\
%      If no methylated cluster arise, then OK to conclusion.
%  \end{itemize}
%\end{frame}
%\begin{frame}{Constitutively open and demethylated regions exhibits wide range of variation in enhancer activity}
%  \begin{itemize}
%    \item 3D + 3E + 3F + 3G
%  \end{itemize}
%\end{frame}
%\begin{frame}{TRAC locus is the highest induced gene during T-Cell committment, yet is constitutively demethylated and open}
%  \begin{itemize}
%    \item 3H + 3H' + S3'
%    \item Seamless transition to mechanistical part about HOXA role...
%  \end{itemize}
%\end{frame}
%\begin{frame}[allowframebreaks]
%  % \printbibliography
%  \bibliographystyle{abbrvnat}
%  %\bibliographystyle{abbrvnat}
%  %\bibliographystyle{dinat}
%  \tiny\bibliography{src/bib/mendeley_collection_2019-02-18}
%\end{frame} 

\end{document}
